src/models/fields/base-field-configuration.ts

```diff
@@ -178,6 +178,12 @@ const schema = (apiVersion: number): Joi.AlternativesSchema => {
       inputType: Joi.string().valid('file', 'files').required(),
       ...deprecatedValueOptions,
     }),
+    commonBaseFieldConfigurationSchema.keys({
+      type: Joi.string().valid('hashId').default('hashId').required(),
+      inputType: Joi.string().valid('pinGroupHashId', 'pinHashId').default('pinGroupHashId').required(),
+      ...deprecatedDefaultValue,
```

Comment:
Why does a new field need deprecated values? Can't we just leave them out?

Reply:
@everhardt maybe you help us here.

Reply:
Doesn't work without it because the client sends a [defaultValue](https://github.com/withthegrid/platform-client/blob/main/src/views/generic/fields/field-configurations.vue#L227). Will check if it can be removed from the frontend without an impact.

I also see this comment:  [// can be dropped after deploy of related platform release](https://github.com/withthegrid/platform-sdk/blob/main/src/models/fields/base-field-configuration.ts#L46). I guess it was left behind and the related platform release happened?

Reply:
> The from the client gets parsed by the platform-sdk models both on client and server. The client side parsing would remove it already I would expect? So if you drop the lines marked with `// can be dropped ...` and only update the server (not the client), I would still expect it to work.

That's true, it does if only the server has the sdk updated

> Also, the defaultValue you link to is for `type: 'string'` and `inputType: 'text'`. There `deprecatedDefaultValue` is not applied and [is still valid](https://github.com/withthegrid/platform-sdk/blob/a61dfba3662c7a92e6e87257e1c6176b439baa2e/src/models/fields/base-field-configuration.ts#L61).

Indeed, it is for `type: 'string'` and `inputType: 'text'`, but `add()` function is called from [here](https://github.com/withthegrid/platform-client/blob/main/src/views/monitoring-environment/settings/field-configurations.vue#L43) and sets `this.dialog.fieldConfiguration` acting as a default configuration which is overwritten, but if defaultValue: '' is not overwritten, it will remain set.

![image](https://user-images.githubusercontent.com/72932961/203975669-9c577540-fd84-4827-a905-d275c7b4ef45.png)

Reply:
Right, so the logic in the client would have to be updated to be able to drop those `deprecatedDefaultValue`'s. Let's leave that for another ticket then.